File: Selenium/Activities/Activity21.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.select import Select
with webdriver.Firefox() as driver:
    wait=WebDriverWait(driver,timeout=10)
    driver.get("https://training-support.net/webelements/tabs")
    print("the title of the page is: " ,driver.title)
    tabButton = driver.find_element(By.CSS_SELECTOR,".bg-purple-200")
    tabButton.click()

    wait.until(EC.number_of_windows_to_be(2))
    print("currently opened windows: ",driver.window_handles)
    for handle in driver.window_handles:
        driver.switch_to.window(handle)

    wait.until(EC.element_to_be_clickable((By.CSS_SELECTOR, ".bg-purple-200")))

    print("Current tab: ", driver.current_window_handle)
    print("New Page title: ", driver.title)
    message = driver.find_element(By.CSS_SELECTOR,".mt-5")
    print("The message on the current page: ",message.text)
    driver.find_element(By.CSS_SELECTOR,".bg-purple-200").click()
    wait.until(EC.number_of_windows_to_be(3))
    print("Currently opened windows:", driver.window_handles)

    for handle in driver.window_handles:
        driver.switch_to.window(handle)

    wait.until(EC.visibility_of_element_located((By.CSS_SELECTOR, ".bg-purple-200")))

    print("Handle of the current page: "+driver.current_window_handle)
    print("The title of the current page: ",driver.title)
    message1 = driver.find_element(By.CSS_SELECTOR,".mt-5")
    print("The message on the current page: ",message1.text)
    driver.quit()
		
		
		

File: Selenium/Activities/Activity21.java
import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity21 {
	public static void main(String args[]) {
		WebDriver driver;
		driver = new FirefoxDriver();
		WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
		
		driver.get("https://training-support.net/webelements/tabs");
		System.out.println("The title of the main page is: "+driver.getTitle());
		WebElement tabButton = driver.findElement(By.cssSelector(".bg-purple-200"));
		tabButton.click();
		
		wait.until(ExpectedConditions.numberOfWindowsToBe(2));
		System.out.println("currently opened windows: "+driver.getWindowHandles());
		
		for(String handle: driver.getWindowHandles()) {
			driver.switchTo().window(handle);
		}
		
		wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector(".bg-purple-200")));
		System.out.println("Handle of current tab: "+driver.getWindowHandle());
		System.out.println("The title of the current page: "+driver.getTitle());
		WebElement message = driver.findElement(By.cssSelector(".mt-5"));
		System.out.println("The message on the current page: "+message.getText());
		driver.findElement(By.cssSelector(".bg-purple-200")).click();
		
		wait.until(ExpectedConditions.numberOfWindowsToBe(3));
		System.out.println("currently opened windows: "+driver.getWindowHandles());
		
		for(String handle:driver.getWindowHandles()) {
			driver.switchTo().window(handle);
		}
		
		wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector(".bg-purple-200")));
		System.out.println("Handle of the current page: "+driver.getWindowHandle());
		System.out.println("The title of the current page: "+driver.getTitle());
		WebElement message1 = driver.findElement(By.cssSelector(".mt-5"));
		System.out.println("The message on the current page: "+message1.getText());
		
		driver.quit();
		
	}

}
